components/src/dynamo/profiler/utils/config_modifiers/protocol.py

```
@@ -779,17 +780,62 @@ def _deep_merge_overrides(
+
+        # Merge env vars by env name. User overrides take precedence over profiler-injected envs.
+        elif (
+            key == "env"
+            and isinstance(value, list)
+            and isinstance(target.get(key), list)
+        ):
+            existing = target.get(key, [])
+            merged: dict[str, dict] = {}
+
+            for env in existing:
+                env_dict = _envvar_to_dict(env)
```

Ruff (0.15.12)

[error] Undefined name `_envvar_to_dict`

(F821)